app/views/play/cinematic/CinematicLankBoss.js

```diff
@@ -17,164 +37,289 @@ const Lank = require('lib/surface/Lank')
-  constructor ({ groundLayer, layerAdapter, camera, loader }) {
+  constructor ({ groundLayer, layerAdapter, backgroundAdapter, camera, loader }) {
     this.groundLayer = groundLayer
-    this.layerAdapter = layerAdapter
+    this.layerAdapters = {
+      'Default': layerAdapter,
+      'Background': backgroundAdapter
+    }
     this.camera = camera
     this.stageBounds = {
       topLeft: this.camera.canvasToWorld({ x: 0, y: 0 }),
       bottomRight: this.camera.canvasToWorld({ x: this.camera.canvasWidth, y: this.camera.canvasHeight })
     }
     this.loader = loader
+    this.lanks = {}
   }
 
   /**
    * Returns a list of commands that correctly set up the shot.
    * @param {Shot} shot - the cinematic shot data.
    */
   parseSetupShot (shot) {
-    const leftCharSlug = getLeftCharacterThangTypeSlug(shot)
     const commands = []
 
-    if (leftCharSlug) {
-      const { slug, enterOnStart, position } = leftCharSlug
-      this.addLank('left', this.loader.getThangType(slug))
+    const moveCharacter = (side, resource, enterOnStart, pos) => {
       if (enterOnStart) {
-        commands.push(this.moveLankCommand('left', position))
+        commands.push(this.moveLankCommand({ key: side, resource, pos }))
       } else {
-        this.moveLank('left', position)
+        commands.push(this.moveLank({ key: side, resource, pos }))
       }
     }
 
+    const background = getBackground(shot)
+    if (background) {
+      commands.push(this.setBackgroundCommand(background))
+    }
+
+    const lHero = getLeftHero(shot)
+    const original = me.get('heroConfig').thangType
+    if (lHero) {
+      const { enterOnStart, thang: { pos } } = lHero
+      moveCharacter('left', original, enterOnStart, pos)
+    }
+
+    const rHero = getRightHero(shot)
+    if (rHero) {
+      const { enterOnStart, thang: { pos } } = rHero
+      moveCharacter('right', original, enterOnStart, pos)
+    }
+
+    const leftCharSlug = getLeftCharacterThangTypeSlug(shot)
+    if (leftCharSlug) {
+      const { slug, enterOnStart, thang: { pos } } = leftCharSlug
+      moveCharacter('left', slug, enterOnStart, pos)
+    }
+
     const rightCharSlug = getRightCharacterThangTypeSlug(shot)
     if (rightCharSlug) {
-      const { slug, enterOnStart, position } = rightCharSlug
-      this.addLank('right', this.loader.getThangType(slug))
-      if (enterOnStart) {
-        commands.push(this.moveLankCommand('right', position))
-      } else {
-        this.moveLank('right', position)
-      }
+      const { slug, enterOnStart, thang: { pos } } = rightCharSlug
+      moveCharacter('right', slug, enterOnStart, pos)
     }
 
     return commands
   }
 
-  registerLank (side, lank) {
-    assertSide(side)
-    this[side] = lank
+  parseDialogNode (dialogNode) {
+    const commands = []
+    const char = exitCharacter(dialogNode)
+    if (char === 'left' || char === 'both') {
+      commands.push(this.moveLankCommand({ key: 'left', pos: { x: -20 } }))
```

Comment:
Currently -20 is a best guess for moving the character off the screen. Need to refactor to use the camera. Works well enough for the illusion atm.
Can add a comment with a `TODO`.